Mathlib/Analysis/CStarAlgebra/Module/Synonym.lean

```diff
@@ -0,0 +1,320 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.RingTheory.Finiteness
+import Mathlib.Topology.Bornology.Constructions
+import Mathlib.Topology.UniformSpace.Equiv
+
+/-! # Type synonym for types with a `CStarModule` structure
+
+It is often the case that we want to construct a `CStarModule` instance on a type that is already
+endowed with a norm, but this norm is not the one associated to its `CStarModule` structure. For
+this reason, we create a type synonym `WithCStarModule` which is endowed with the requisite
+`CStarModule` instance. We also introduce the scoped notation `C⋆ᵐᵒᵈ` for this type synonym.
```

Comment:
I'm still very confused here; where is it "endowed with the requisite
`CStarModule` instance"? A type synonym with no new instances strikes me as quite weird.